Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -97,6 +96,13 @@ public enum BoundingBoxActivationType
             ActivateByProximityAndPointer,
             ActivateManually
         }
+
+        private enum HandleProximityState
```

Comment:
Is this something we could reuse somewhere?

Reply:
I was also actually hoping that instead of modifying boundingbox we could create a  new ProximityHandler component that would expose an event like OnPointerProximityChanged(ProximityEventData d), as well as a way to poll for the closest pointer information (which would return Infinity of no pointer is within grabbing distance). You would also probably want to have events like OnNearStateEntered, OnFarStateEntered, OnMediumStateEntered.

ProximityEventData would be something like this:

```
class ProximityEventData
{
    ProximityState oldState;
    ProximityState newState;
    float oldDistance;
    float newDistance;
}

enum ProximityState
{
    FarDistance,
    NearDistance,
    MediumDistance
}
```

I think that with that component you could build the behavior you need for the bounding box handles by adding a ProximityHandler to each of the corner prefabs. 

You could then just listen for those proximity events and change the visuals in the way you're doing here. Or, a designer could hook up the Unity events OnNearStateEntered, etc. to set different states in an Interactable, which would allow for somebody to make really interesting visual feedback based on the proximity of a hand to a corner.

Reply:
that is a major architecture change. the bb currently centralizes the handle grabbing test in order to avoid a script running on each handle. At this point- with the evolution of the bb from the simple tinkertoy style to the Shell Parity functionality- i think the handle-based script is a great idea but i think its beyond the scope of this bug fix.?

Reply:
Yes it's probably out of scope, though it would be good to create an issue so we don't forget about this suggestion. We can then decide if that issue is in scope for GA. I created #4689 to track this

Reply:
cool!